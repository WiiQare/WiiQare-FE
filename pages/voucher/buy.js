import { createContext, useState } from 'react';
import Head from 'next/head';
import { useRouter } from 'next/router';
import DashboardLayout from '../../layouts/Dashboard';
import CardHeader from '../../components/atoms/Card/Header';
import Stepper from '../../components/atoms/Stepper';
import Step from '../../components/atoms/Stepper/step';
export const FormContext = createContext();

const Page = () => {
  const { step, redirect_status, payment_intent } = useRouter().query;
  const [activeStepIndex, setActiveStepIndex] = useState(
    step == 'end' && redirect_status == 'succeeded' ? 2 : 0,
  );
  const [formData, setFormData] = useState({});

  return (
    <>
      <Head>
        <title>Envoyer un pass santé</title>
      </Head>
      <div className="p-2 space-y-6 md:py-8 md:px-6">
        <CardHeader
          title={'Envoyer un pass santé'}
          breadcrumbs={[
            {
              item: 'Accueil',
              link: '/',
            },
            {
              item: 'Envoyer un pass santé',
              link: '/voucher/buy',
            },
          ]}
          download={false}
          print={false}
        />

        <section className="w-full pb-20 md:pb-0">
          <div className="w-full overflow-hidden md:col-span-2 rounded-lg py-8 flex flex-col gap-6 bg-white drop-shadow-sm">
            <div className="px-8 space-y-8">
              <div>
                <h2 className="font-semibold text-3xl">
                  Envoyer un pass santé
                </h2>
              </div>
              <div className="flex flex-col gap-4">
                <FormContext.Provider
                  value={{
                    activeStepIndex,
                    setActiveStepIndex,
                    formData,
                    setFormData,
                    payment_intent,
                  }}
                >
                  <div className="flex flex-col items-center justify-start">
                    <Stepper />
                    <Step />
                  </div>
                </FormContext.Provider>
              </div>
            </div>
          </div>
        </section>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;
