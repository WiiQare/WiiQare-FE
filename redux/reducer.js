import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  client: { register: {}, patient: {} },
};

export const ReducerSlice = createSlice({
  name: 'wiiqare',
  initialState,
  reducers: {
    setRegister: (state, action) => {
      state.client.register = { ...action.payload };
    },
    setPatientDispatch: (state, action) => {
      state.client.patient = { ...action.payload };
    },
  },
});

export const { setRegister, setPatientDispatch } = ReducerSlice.actions;

export default ReducerSlice.reducer;
