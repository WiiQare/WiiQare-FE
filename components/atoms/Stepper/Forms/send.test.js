import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Send from './send';
import { FormContext } from '../../../../pages/voucher/buy';
import { SessionProvider } from 'next-auth/react';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/store';
import { QueryClientProvider, QueryClient } from 'react-query';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {
      'payment-intent': 'pi_1J4JrjGswQjYFZwX0Z1Z1Z1Z',
    },
  }),
}));

describe('Send', () => {
  let component;
  beforeEach(() => {
    console.log = jest.fn();
    const queryClient = new QueryClient();
    const res = render(
      <SessionProvider session={{ user: { data: { userId: 'random123' } } }}>
        <QueryClientProvider client={queryClient}>
          <Provider store={store}>
            <FormContext.Provider value={{ activeStepIndex: 0 }}>
              <Send />
            </FormContext.Provider>
          </Provider>
        </QueryClientProvider>
      </SessionProvider>,
    );
    component = res.container;
  });

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
