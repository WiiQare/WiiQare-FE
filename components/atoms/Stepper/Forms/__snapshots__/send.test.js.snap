// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send should render the component 1`] = `
<div>
  <div
    class="flex flex-col gap-6 justify-center items-center"
  >
    <div
      class="relative"
    >
      <div
        class="flex flex-col items-center text-center space-y-2"
      >
        <div
          class="flex flex-col items-center select-none"
        >
          <img
            alt="logo"
            class="h-6 md:h-9 object-left object-contain w-min"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
          <h1
            class="font-extrabold text-gray-700 text-lg hidden md:flex"
          >
            Pass Sante
          </h1>
        </div>
        <span
          class="text-xs flex items-center gap-1"
        >
          Pass Sante ID:
          <div
            class="flex items-center gap-1"
          >
            [
            <div
              class="tooltip"
              data-tip="Copy to clipboard"
            >
              <span
                class="text-orange cursor-pointer"
              >
                0xcda147
                ...
                f368eb4
              </span>
            </div>
            ]
          </div>
        </span>
      </div>
      <div
        class="flex flex-col items-center gap-4"
      >
        <div
          class="border relative border-gray-300 rounded-lg overflow-hidden"
        >
          <div>
            QRCode
          </div>
        </div>
        <div
          class="flex flex-col items-center gap-1"
        >
          <div
            class="flex -space-x-2"
          >
            <img
              alt="Image Description"
              class="inline-block h-[2.875rem] w-[2.875rem] rounded-full ring-2 ring-white dark:ring-gray-800"
              src="/images/homme.png"
            />
            <img
              alt="Image Description"
              class="inline-block h-[2.875rem] w-[2.875rem] rounded-full ring-2 ring-white dark:ring-gray-800"
              src="/images/femme.png"
            />
          </div>
          <h4
            class="text-sm text-center"
          >
            <span
              class="font-semibold"
            >
              $ 0.00
            </span>
             Health Pass WiiQare 
            <br />
             From
             
            <span
              class="text-orange font-semibold"
            >
              0xcda147
              ...
              f368eb4
            </span>
             
            To
             
            <span
              class="text-orange font-semibold"
            >
              0xcda147
              ...
              f368eb4
            </span>
          </h4>
        </div>
      </div>
      <div
        class="text-center mt-6 flex flex-col gap-2 space-y-3"
      >
        <h4
          class="font-semibold text-gray-700 text-sm"
        >
          Envoyer le pass santé au bénéficiaire:
        </h4>
        <div
          class="flex justify-between"
        >
          <a
            class="text-gray-900 bg-white hover:bg-gray-100 border border-gray-200 focus:ring-4 focus:outline-none focus:ring-gray-100 font-medium rounded-lg text-sm px-5 py-2.5 text-center inline-flex items-center gap-2 mr-2 mb-2"
            href="whatsapp://send?text=undefined/voucher/pass/0xcda1470a8117daaccf368eb4"
          >
            <img
              alt=""
              class="w-6"
              src="/images/whatsapp.png"
            />
            <span
              class="hidden md:flex"
            >
              WhatsApp
            </span>
          </a>
          <a
            class="text-gray-900 bg-white hover:bg-gray-100 border border-gray-200 focus:ring-4 focus:outline-none focus:ring-gray-100 font-medium rounded-lg text-sm px-5 py-2.5 text-center inline-flex items-center gap-2 mr-2 mb-2"
            href="https://www.facebook.com/share.php?u=undefined/voucher/pass/0xcda1470a8117daaccf368eb4"
          >
            <img
              alt=""
              class="w-6"
              src="/images/facebook-share.png"
            />
            <span
              class="hidden md:flex"
            >
              Facebook
            </span>
          </a>
          <a
            class="text-gray-900 bg-white hover:bg-gray-100 border border-gray-200 focus:ring-4 focus:outline-none focus:ring-gray-100 font-medium rounded-lg text-sm px-5 py-2.5 text-center inline-flex items-center gap-2 mr-2 mb-2"
            href="sms://+243814978651&?body=undefined/voucher/pass/0xcda1470a8117daaccf368eb4"
          >
            <img
              alt=""
              class="w-6"
              src="/images/sms.png"
            />
            <span
              class="hidden md:flex"
            >
              Message
            </span>
          </a>
          <button
            class="text-gray-900 bg-white hover:bg-gray-100 border border-gray-200 focus:ring-4 focus:outline-none focus:ring-gray-100 font-medium rounded-lg text-sm px-5 py-2.5 text-center inline-flex items-center gap-2 mr-2 mb-2"
            type="button"
          >
            <img
              alt=""
              class="w-6"
              src="/images/text.png"
            />
            <span
              class="hidden md:flex"
            >
              Copy Link
            </span>
          </button>
        </div>
      </div>
      <div
        class="flex-shrink-0 border-none backdrop-filter backdrop-blur-[6px] w-full h-full bg-white/30 absolute top-0 p-6"
      />
      <div
        class="flex flex-shrink-0 absolute top-0 justify-center items-center w-full h-full"
      >
        <div
          class="flex flex-col gap-8 justify-center items-center"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="150"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="150"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
              fill-rule="evenodd"
            />
          </svg>
          <button
            class="bg-orange flex gap-2 effect-up justify-center items-center text-gray-100 font-normal h-fit  py-2 px-3 rounded-lg text-sm transition duration-300"
          >
            <svg
              aria-hidden="true"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
             Voir Pass Santé
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
